changelogs/unreleased/gh-5924-graceful-shutdown.md

```diff
@@ -0,0 +1,40 @@
+## feature/iproto
+
+* Introduce `graceful shutdown` mechanism
+* Introduce `net.box.connection:shutdown()` - send `IPROTO_SHUTDOWN` to the server
+* Introduce `net.box.connection:set_shutdown_handler(handler)` - set handler to do somethink after getting `IPROTO_SHUTDOWN` from the server and sending `IPROTO_SHUTDOWN`  in response to the server
+* Introduce new boolean value of `_session_settings` space - `_session_settings.graceful_shutdown`
+
+Connection support graceful shutdown if it set `_session_settings.graceful_shutdown = true`. By default it is `false` and connection doesn't support graceful shutdown.
+
+### Common flow for shutdown:
```

Comment:
I think there is no need to write all information from the task here. Only what is done. @Korablev77 what do you think about it?

Reply:
This is not a copy of the task, this is a description of how everything should work in my implementation(but it is similar to the task since I took its text as a basis and changed it).